### PR TITLE
[New-parser] parser_mpirun_version

### PR DESCRIPTION
**Pull request by mohitkumarrh**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

*Adding a Specs for Advisory rule mpirun_version*

### CODE REVIEW

**Comment by psachin** [comment]
@mohitkumarrh 

The output collected by the command has to be collected in the file. Please add an entry in https://github.com/RedHatInsights/insights-core/blob/master/insights/specs/insights_archive.py. Use the mangle  command (https://github.com/RedHatInsights/insights-core/blob/master/insights/specs/insights_archive.py) to generate the file_path.

**Comment by psachin** [comment]
Please also create a file in https://github.com/RedHatInsights/insights-core/tree/master/docs/shared_parsers_catalog for documentation.

**Comment by mohitkumarrh** [comment]
@psachin I have added an entry in insights_archive.py. I'll add the documentation too.

File: insights/specs/insights_archive.py
Comment (psachin) [inline insights/specs/insights_archive.py:176]: @mohitkumarrh This should be `insights_commands/usr.local.bin.mpich_--version`


```
$ mangle "/usr/local/bin/mpich --version"
insights_commands/usr.local.bin.mpich_--version
```

Comment (mohitkumarrh) [reply insights/specs/insights_archive.py]: @psachin - Thanks for the review, I have updated the MR.

File: docs/shared_parsers_catalog/mpirun_ver.rst
Comment (psachin) [inline docs/shared_parsers_catalog/mpirun_ver.rst:1]: The filename should be the same as the parser filename i.e `mpirun.rst`

Comment (psachin) [reply docs/shared_parsers_catalog/mpirun_ver.rst:1]: You can optionally run `make html` from `docs/` to check the generate documentation related to the parser.

File: insights/parsers/mpirun.py
Comment (psachin) [inline insights/parsers/mpirun.py:24]: This is not rendered correctly. How about,

```
    ``mpirun`` versions and year mapping is as follow:

        - openmpi-3.1.0 to openmpi-3.1.3 -> 2018
        - openmpi-3.1.4 to openmpi-3.1.6 -> 2019
        - openmpi-4.0.0 to openmpi-4.1.0 -> 2020
        - openmpi-4.1.1 to openmpi-4.1.2 -> 2021
        - openmpi-5.0.0 - continue_to_release -> 2022
```

File: insights/parsers/mpirun.py
Comment (psachin) [inline insights/parsers/mpirun.py:37]: Better break this line into two separate exceptions:

1.
```
if not content:
    raise SkipException("Empty content")
```

2.
```
if len(content) > 2:
    raise SkipException("Conent not parsable")
```

Something like that.

**Comment by mohitkumarrh** [comment]
@psachin - Thanks for the review. I have made the suggested changes, Please review.
py.test:
```
(venv) ➜  insights-core git:(parser_mpirun_version) ✗ py.test -svk mpirun --cov=insights.parsers.mpirun
============================================================= test session starts =============================================================
platform darwin -- Python 3.6.5, pytest-5.2.1, py-1.10.0, pluggy-0.13.1 -- /Users/mohitkumar/Insights/venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/mohitkumar/Insights/insights-core, inifile: setup.cfg, testpaths: insights
plugins: cov-2.12.1
collecting 181 items                                                                                                                          Downloading a fresh and hot uploader.json...
collected 2875 items / 2872 deselected / 3 selected                                                                                           

insights/tests/parsers/test_mpirun.py::test_mpirun_version PASSED
insights/tests/parsers/test_mpirun.py::test_mpirun_version_ab PASSED
insights/tests/parsers/test_mpirun.py::test_doc_examples PASSED

---------- coverage: platform darwin, python 3.6.5-final-0 -----------
Name                         Stmts   Miss Branch BrPart  Cover   Missing
------------------------------------------------------------------------
insights/parsers/mpirun.py      12      0      6      0   100%
------------------------------------------------------------------------
TOTAL                           12      0      6      0   100%


===================================================== 3 passed, 2872 deselected in 7.87s ======================================================
```

File: insights/tests/parsers/test_mpirun.py
Comment (psachin) [inline insights/tests/parsers/test_mpirun.py:43]: This test is skipped because you don't have a check for this.

File: insights/tests/parsers/test_mpirun.py
Comment (psachin) [inline insights/tests/parsers/test_mpirun.py:47]: SInce there are different exception warnings you can check the content instead of checking the return values. Please refer https://github.com/RedHatInsights/insights-core/blob/master/insights/tests/parsers/test_subscription_manager_release.py#L66

Comment (mohitkumarrh) [reply insights/tests/parsers/test_mpirun.py:47]: Ack, I'll update the PR accordingly

**Comment by mohitkumarrh** [comment]
@psachin please review
```
➜  insights-core git:(parser_mpirun_version) ✗ py.test -svk mpirun --cov=insights.parsers.mpirun
=================================================== test session starts ====================================================
platform darwin -- Python 3.9.1, pytest-7.1.2, pluggy-1.0.0 -- /Users/mohitkumar/.pyenv/versions/3.9.1/bin/python3.9
cachedir: .pytest_cache
rootdir: /Users/mohitkumar/Insights/insights-core, configfile: setup.cfg, testpaths: insights
plugins: cov-3.0.0
collecting 181 items                                                                                                       Downloading a fresh and hot uploader.json...
collected 2875 items / 2872 deselected / 3 selected                                                                        

insights/tests/parsers/test_mpirun.py::test_mpirun_version PASSED
insights/tests/parsers/test_mpirun.py::test_mpirun_version_ab PASSED
insights/tests/parsers/test_mpirun.py::test_doc_examples PASSED

---------- coverage: platform darwin, python 3.9.1-final-0 -----------
Name                         Stmts   Miss Branch BrPart  Cover   Missing
------------------------------------------------------------------------
insights/parsers/mpirun.py      14      0      8      0   100%
------------------------------------------------------------------------
TOTAL                           14      0      8      0   100%


============================================ 3 passed, 2872 deselected in 6.14s ============================================
```

File: insights/specs/insights_archive.py
Comment (bfahr) [inline insights/specs/insights_archive.py:176]: @mohitkumarrh looks like you have a typo in the spec name.  The `mangle` command provides the correct path in an archive:
```
$ mangle "/usr/local/bin/mpirun --version"
insights_commands/usr.local.bin.mpirun_--version
```

File: insights/specs/default.py
Comment (bfahr) [inline insights/specs/default.py:405]: Is this the expected location of this command?  Are there other typical locations where this could be installed?  In the parser docstring you specify `/usr/bin/mpirun`.  If it may be located in either location then you can use the `first_of` spec, like [this](https://github.com/RedHatInsights/insights-core/blob/master/insights/specs/default.py#L367).

**Review by bfahr — CHANGES_REQUESTED** [review]
Also, this spec requires approval before this PR can be merged.

File: insights/specs/insights_archive.py
Comment (mohitkumarrh) [reply insights/specs/insights_archive.py:176]: @bfahr - Thanks for the review, Yes i made a typo mistake in mangle command. I'll correct this typo and update the PR

**Comment by xiangce** [comment]
> Also, this spec requires approval before this PR can be merged.

@mohitkumarrh - as mentioned by Bob, Spec approval is required before merging this parser, I cannot find any MR/issues for it in the insights-core-assets repo, would you please open one for it?

**Comment by mohitkumarrh** [comment]
@xiangce - As you and Bob mentioned, I have raise a PR for Spec approval.

MR/issues for it in the insights-core-assets repo: https://gitlab.cee.redhat.com/insights-release-eng/insights-core-assets/-/merge_requests/474

@psachin @xiangce - please have a final review

**Review by psachin — APPROVED** [review]

File: insights/specs/default.py
Comment (psachin) [reply insights/specs/default.py:405]: This makes sense to me because recently I noticed that the generic openmpi package install the binary in ` /usr/lib64/openmpi/bin/mpirun`

**Comment by mohitkumarrh** [comment]
@xiangce - Hey as we discussed today, I have made all the changes in PR you asked me to make, please have a Review now.

**Comment by mohitkumarrh** [comment]
py.test:
```
(venv) ➜  insights-core git:(parser_mpirun_version) ✗ py.test -svk mpirun --cov=insights.parsers.mpirun
========================================================= test session starts ==========================================================
platform darwin -- Python 3.6.5, pytest-5.2.1, py-1.10.0, pluggy-0.13.1 -- /Users/mohitkumar/Insights/venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/mohitkumar/Insights/insights-core, inifile: setup.cfg, testpaths: insights
plugins: cov-2.12.1
collecting 181 items                                                                                                                   Downloading a fresh and hot uploader.json...
collected 2875 items / 2872 deselected / 3 selected                                                                                    

insights/tests/parsers/test_mpirun.py::test_mpirun_version PASSED
insights/tests/parsers/test_mpirun.py::test_mpirun_version_ab PASSED
insights/tests/parsers/test_mpirun.py::test_doc_examples PASSED

---------- coverage: platform darwin, python 3.6.5-final-0 -----------
Name                         Stmts   Miss Branch BrPart  Cover   Missing
------------------------------------------------------------------------
insights/parsers/mpirun.py      12      0      6      0   100%
------------------------------------------------------------------------
TOTAL                           12      0      6      0   100%


================================================== 3 passed, 2872 deselected in 7.56s ==================================================
```

**Review by xiangce — APPROVED** [review]

**Review by xiangce — COMMENTED** [review]
Just noticed that you have named the doc entry file as `mpirunversion.rst`, is it a mistake or on your purpose?

**Comment by mohitkumarrh** [comment]
> 

@xiangce - I renamed the file by mistake, I have update the file name please have a look

File: insights/specs/default.py
Comment (mohitkumarrh) [reply insights/specs/default.py:405]: @psachin if we will use the `usr/lib64/openmpi/bin/mpirun` binary to get the version, the out we get is totally different i.e. mpirun (Open MPI) 4.1.1
from 4.1.1 we can't detect the year as we don't have correct mapping. so i think we can enhance the rule later, but it is currently sufficient for the jira i'm working on.

Comment (psachin) [reply insights/specs/default.py:405]: Ok @mohitkumarrh I think we can merge the changes for now.

**Pull request merged**
